tapdb/universe_federation.go

```diff
@@ -174,4 +220,190 @@ func (u *UniverseFederationDB) LogNewSyncs(ctx context.Context,
 	})
 }
 
-var _ universe.FederationLog = (*UniverseFederationDB)(nil)
+// UpsertFederationSyncConfig upserts both the global and universe specific
+// federation sync configs.
+func (u *UniverseFederationDB) UpsertFederationSyncConfig(
+	ctx context.Context, globalSyncConfigs []*universe.FedGlobalSyncConfig,
+	uniSyncConfigs []*universe.FedUniSyncConfig) error {
+
+	var writeTx UniverseFederationOptions
+	return u.db.ExecTx(ctx, &writeTx, func(db UniverseServerStore) error {
+		// Upsert global proof type specific federation sync
+		// configs.
+		for i := range globalSyncConfigs {
+			config := globalSyncConfigs[i]
+
+			params := UpsertFedGlobalSyncConfigParams{
+				ProofType:       config.ProofType.String(),
+				AllowSyncInsert: config.AllowSyncInsert,
+				AllowSyncExport: config.AllowSyncExport,
+			}
+			err := db.UpsertFederationGlobalSyncConfig(ctx, params)
+			if err != nil {
+				return err
+			}
+		}
+
+		// Upsert universe specific sync configs.
+		for i := range uniSyncConfigs {
+			var (
+				config = uniSyncConfigs[i]
+
+				uniID        = config.UniverseID
+				groupPubKey  []byte
+				assetIDBytes []byte
+			)
+
+			// If the group key is set, then we'll serialize it
+			// into bytes.
+			if uniID.GroupKey != nil {
+				groupPubKey = uniID.GroupKey.SerializeCompressed()
+			} else {
+				// If the group key is not set, then we'll use
+				// the asset ID. The group key supersedes the
+				// asset ID.
+				assetIDBytes = uniID.AssetID[:]
+			}
+
+			err := db.UpsertFederationUniSyncConfig(
+				ctx, UpsertFedUniSyncConfigParams{
+					AssetID:         assetIDBytes,
+					GroupKey:        groupPubKey,
+					ProofType:       uniID.ProofType.String(),
+					AllowSyncInsert: config.AllowSyncInsert,
+					AllowSyncExport: config.AllowSyncExport,
+				},
+			)
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+}
+
+// QueryFederationSyncConfigs returns the global and universe specific
+// federation sync configs.
+func (u *UniverseFederationDB) QueryFederationSyncConfigs(
+	ctx context.Context) ([]*universe.FedGlobalSyncConfig,
+	[]*universe.FedUniSyncConfig, error) {
+
+	var (
+		readTx UniverseFederationOptions
+
+		globalConfigs []*universe.FedGlobalSyncConfig
+		uniConfigs    []*universe.FedUniSyncConfig
+	)
+
+	err := u.db.ExecTx(ctx, &readTx, func(db UniverseServerStore) error {
+		// Query for global sync configs.
+		globalDbConfigs, err := db.QueryFederationGlobalSyncConfigs(
+			ctx,
+		)
+		switch {
+		case errors.Is(err, sql.ErrNoRows):
+			// If the query does not return any rows then a global
+			// config for each proof type has not yet been set. We
+			// will return a default config for each proof type.
+			globalConfigs = []*universe.FedGlobalSyncConfig{
+				{
+					ProofType:       universe.ProofTypeIssuance,
+					AllowSyncInsert: true,
```

Comment:
This needs to be reconciled with the command line args we have. Perhaps we eliminate them and have them map here instead? 

Eg: rn the default is to not allow inserting if not specified (even for issuance), but then this flips that.

Reply:
In the latest commits, I've modified such that we default to not allow globally for issuance and transfer proof universes.